examples/fabric/build_your_own_trainer/trainer.py

```diff
@@ -156,6 +156,20 @@ def fit(
         assert optimizer is not None
         model, optimizer = self.fabric.setup(model, optimizer)
 
+        if not is_overridden("on_validation_model_eval", _unwrap_objects(model)):
```

Comment:
@liambsmith I would prefer if we just call `.eval()` and `.train()` directly on the model. In the Fabric framework, that's normally the expectation. 

Since the Build-your-own-trainer is an example here, I would keep it lean and easy to understand.

Reply:
> @liambsmith I would prefer if we just call `.eval()` and `.train()` directly on the model. In the Fabric framework, that's normally the expectation.
> 
> Since the Build-your-own-trainer is an example here, I would keep it lean and easy to understand.

I have changed the code to do that instead, does that work better? This keeps the simpler code of the example while still fixing the error.

Reply:
Thanks!